### ISSUE TITLE
Mergestruct bug in matlab version 2024a

### ISSUE THREAD

**Issue by JoaoAmaro2001** [fieldtrip/fieldtrip#2420]
**Describe the bug**

I get the following error when trying to run data2bids on matlab 2024a to "bidify" eeg data:
```matlab
Error using mergestruct
Too many input arguments.

Error in data2bids (line 1653)
  coordsystem_json = mergestruct(coordsystem_settings, coordsystem_json, false); % FIXME the order of precedence is different here
```
The script runs by taking the "false" input out of all instances of the mergestruct function. I have not tested this function in other matlab versions.

**Environment and versions**
 - Operating System: Operating System: Microsoft Windows 11 Enterprise Version 10.0 (Build 22631)
 - MATLAB Version: 24.1.0.2568132 (R2024a) Update 1
 - FieldTrip version 20240502.

**Comment by robertoostenveld** [comment]
which fieldtrip version are you using? Can you check which `mergestruct` function is being called by setting a MATLAB debug breakpoint on line 1653 and then typing

```
which mergestruct
```

It should be `fieldtrip/private/mergestruct.m` which takes three input arguments (s1, s2, emptymeaningful).

**Comment by JoaoAmaro2001** [comment]
When I restarted matlab it suddenly started working. For some reason sometimes I encounter errors due to incorrect path settings even though I always set it up as suggested by fieldtrip:
```matlab
restoredefaultpath
addpath('C:\toolbox\fieldtrip-20240502')
ft_defaults
```
It seems the external folder which is added by data2bids was not being added the first time. After encountering this problem, I copied mergestruct.m to another folder and data2bids.m was now using this copied version. After deleting this copied version and restarting matlab everything went well.

Anyways, running `which mergestruct` gave me the following path, which seems to be the correct one:
C:\toolbox\fieldtrip-20240502\external\jsonlab\mergestruct.m

Thank you for your time!

**Comment by JoaoAmaro2001** [comment]
When I restarted matlab it suddenly started working. For some reason sometimes I encounter errors due to incorrect path settings even though I always set it up as suggested by fieldtrip:
```matlab
restoredefaultpath
addpath('C:\toolbox\fieldtrip-20240502')
ft_defaults
```
It seems the external folder which is added by data2bids was not being added but it is working now. After encountering this problem I copied mergestruct.m to another folder

Anyways, running `which mergestruct` gave me the following path, which seems to be the correct one:
C:\toolbox\fieldtrip-20240502\external\jsonlab\mergestruct.m

Thank you for your time!
